src/module_3/push_exploration.md

```diff
@@ -0,0 +1,5364 @@
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>Feature</th>
+      <th>Coefficient</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>2</th>
+      <td>user_id</td>
+      <td>-0.017418</td>
+    </tr>
+    <tr>
+      <th>7</th>
+      <td>count_pets</td>
+      <td>0.016740</td>
+    </tr>
+    <tr>
+      <th>44</th>
+      <td>product_feedingweaning</td>
+      <td>-0.014295</td>
+    </tr>
+    <tr>
+      <th>19</th>
+      <td>product_babyfood12months</td>
+      <td>0.009898</td>
+    </tr>
+    <tr>
+      <th>8</th>
+      <td>people_ex_baby</td>
+      <td>-0.009382</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>order_id</td>
+      <td>0.008983</td>
+    </tr>
+    <tr>
+      <th>16</th>
+      <td>day_of_week</td>
+      <td>0.006718</td>
+    </tr>
+    <tr>
+      <th>70</th>
+      <td>product_superfoodssupplements</td>
+      <td>-0.002507</td>
+    </tr>
+    <tr>
+      <th>67</th>
+      <td>product_shavinggrooming</td>
+      <td>-0.000317</td>
+    </tr>
+    <tr>
+      <th>6</th>
+      <td>global_popularity</td>
+      <td>0.000000</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+- We see any event account for 0.5 in the model (remember we created this variable). We could undo it and introduce the full binary cols, as they seem very relevant
+- We see even people_ex_baby was not used, despite we kept the most representative one. We could delete it
+- We see day_of_week throwed no info, we could instead present day_of_month
+- We see global_popularity was forced to 0, so we can delete it also
+
+
+```python
+binary = ['ordered_before', 'abandoned_before', 'active_snoozed', 'set_as_regular']
+
+cols = information + numerical + categorical + binary + predicted
+final_data = filtered_data[cols]
+
+#Verify created at has pandas date_format
+final_data['created_at'] = pd.to_datetime(final_data['created_at'])
+
+# Extract year, month, day, and hour as separate features
+final_data['year'] = final_data['created_at'].dt.year
+final_data['month'] = final_data['created_at'].dt.month
+# Extract the day of the week (numerical, 1-7, starting with Monday as 1)
+final_data['day_of_month'] = final_data['created_at'].dt.day
+final_data['hour'] = final_data['created_at'].dt.hour
+final_data = final_data.drop(columns=['created_at'])
+
+# Apply one-hot encoding for 'vendor'
+# Drop_first = True to get rid of an additional col. Binaries for keeping consistency as any_event
+final_data = pd.get_dummies(final_data, columns=['product_type'], prefix='product', drop_first=True).astype(int)
+
+#Drop cols
+final_data = final_data.drop(columns=['global_popularity', 'people_ex_baby'])
+
+print(final_data.shape)
+final_data.head()
+```
+
+    /tmp/ipykernel_481/2320102903.py:7: SettingWithCopyWarning: 
```

Comment:
Same as before, avoid the warnings by doing: `final_data = filtered_data[cols].copy()`